src/main/java/com/google/sps/servlets/PlaceGuideServlet.java

```diff
@@ -199,8 +201,10 @@ private PlaceGuide getPlaceGuideFromRequest(HttpServletRequest request) {
     if (imageKey != null) {
       newPlaceGuideBuilder.setImageKey(imageKey);
     } else if (request.getParameterValues(DELETE_IMAGE_INPUT) == null) {
-      imageKey = placeGuideRepository.getPlaceGuide(id).getImageKey();
-      newPlaceGuideBuilder.setImageKey(imageKey);
+      if (!newPlaceGuide) {
+        PlaceGuide prevPlaceGuide = placeGuideRepository.getPlaceGuide(id);
+        newPlaceGuideBuilder.setImageKey(prevPlaceGuide.getImageKey());
```

Comment:
Can we rename "newPlaceGuideBuilder" to "placeGuideBuilder"? Since this might not be a new one.

Reply:
done.